client/core/trade_simnet_test.go

```diff
@@ -438,7 +438,7 @@ func monitorTradeForTestOrder(ctx context.Context, client *tClient, orderID stri
 			// up action. Mining now without waiting may cause the other client
 			// to perform the follow up action before the goroutine captures
 			// this status change.
-			time.Sleep(1 * time.Second)
+			time.Sleep(5 * time.Second)
```

Comment:
I've noticed scenarios where client A sends a swap or redeem tx, but the other client's trade monitoring goroutine does not report the status change until about 4 seconds later.

This is particularly problematic if we want the other client to NOT perform the expected follow up action, such as taker not sending counter swap after receiving notice of maker's swap. If sufficient time is not allowed for the taker's trade monitoring goroutine to capture the `MakerSwap` status before the maker's swap is mined, the taker might recognize that maker's swap has gotten the required confs and proceed to send a counterswap before the trade monitoring routine tries to prevent further action.

Reply:
Now that I think about it, instead of waiting for 5 seconds which may be more than necessary in some instances, another thought is to move the mining action into the other client's trade monitoring goroutine. That ensures that party A's swap or redeem is only mined after party B's trade monitoring goroutine has had the chance to handle the status change. In cases where party B should not follow up, we set `trade.failErr` on party B's trade before mining party A's tx.

Will give it a try and push if it doesn't over-complicate the code flow.

Basically, party B cannot / should not act until party A's tx gets the required confs, so we only mine party A's tx when we're ready for party B to perform its follow up action. The exception to this is Taker redeeming Maker's swap without needing to wait for Maker's redeem to get any confs.